src/main/java/org/interledger/cryptoconditions/ConditionImpl.java

```diff
@@ -34,21 +46,63 @@ public ConditionImpl(ConditionType type, EnumSet<FeatureSuite> features, byte[]
 		this.maxFulfillmentLength = maxFulfillmentLength;
 	}
 
+	public ConditionImpl(String URI) {
```

Comment:
We should have this logic in a ConditionReader rather than a constructor. This is too much processing for a constructor.